app/admin.rb

```diff
@@ -3,24 +3,37 @@
 module Admin
   extend Discordrb::Commands::CommandContainer
 
-  command :adddeath, help_available: false do |event, *parse_string|
+  def self.find_role(event, role)
     role = nil
-    parse_string = parse_string.join(' ')
-    parse_string = parse_string.split('-', 2)
-    name = parse_string[0].chomp
-    reason = parse_string[1].chomp.gsub!(/^\s+/, '')
-
-    puts name
-    puts reason
-
     event.server.roles.each do |server_role|
       next unless server_role.name == 'botadmin'
       role = server_role
-      break
     end
+    return role
+  end
 
+  command :updateprofile, help_available: false do |event|
+    role = find_role(event, 'botadmin')
     break unless event.user.role?(role)
```

Comment:
Minor issue, but since a review was asked... to me it looks like there's an unnecessary role variable here, I would do something like
`break unless event.user.role?(find_role(event, 'botadmin'))
`
but not having ever coded a single line of ruby, I don't know if that works. Same with line 29-30.

If there were a lot of roles, and a lot of find_role's going on, it's inefficient to always loop though all of them. Instead, an efficient check could be made that just checks if a single role is present, and the role is cached. In reality this doesn't really matter at all, but if this was to become a bot that's on a thousand discord servers with tens of thousands of users, it could be something to consider. Perhaps even the ruby discord library has some efficent way of doing this already.

Then one thing I shall have to mention is that there's a LOT of hardcoding going on here. I would at least add the 'botadmin' hardcoded string to a variable so it could be changed from a single place instead of search-replacing it from multiple places. That could even be read from a config file so it can be changed without restarting the bot.
Still, a :+1: from me, good to merge.